aws-ebs-csi-driver/templates/deployment.yaml

```diff
@@ -0,0 +1,147 @@
+# Controller Service
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: ebs-csi-controller
+  namespace: kube-system
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      app: ebs-csi-controller
+      app.kubernetes.io/name: {{ include "aws-ebs-csi-driver.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app: ebs-csi-controller
+        app.kubernetes.io/name: {{ include "aws-ebs-csi-driver.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+      {{- if .Values.podAnnotations }}
+      annotations:
+{{ toYaml .Values.podAnnotations | indent 8 }}
+      {{- end }}
+    spec:
+      {{- with .Values.nodeSelector }}
```

Comment:
I added nodeSelector to the deployment as I have several pools of nodes available and wanted to force this to run in one of them.  I didn't add it to the daemonset as I actually want that to run on all the nodes (and it didn't seem like nodeSelector would be the best way to limit the daemonset's scheduling in general) in all my pools so that I can create dynamic volumes from any of my resources.  Unless I am misunderstanding the relationship between controller service and the node service.  Does all that make sense?

Reply:
If it helps any the preferred values is in the default values file so unless someone explicitly overrides it they will still get the selector `beta.kubernetes.io/os: linux`

https://github.com/kubernetes-sigs/aws-ebs-csi-driver/pull/423/commits/d37cdac6974ee51a45e5c33fdabd5a78ffc9af27#diff-f321d032c2aea6ec0e1651b6b60e565fR40

Reply:
I don't know why I wasn't remembering that nodeSelector is a map.  I changed `beta.kubernetes.io/os: linux` back to being hardcoded for node selection and kept the ability to add addition key/value pairs as node selectors as well.  So, someone can add to the map but they can't remove the `beta.kubernetes.io/os: linux` requirement.

Reply:
For some reason I wasn't remembering that nodeSelector is a map.  I have added back `beta.kubernetes.io/os: linux` as a hardcoded selector so it can't be overridden, but other selectors can still be added.